src/DependencyInjection/SonataUserExtension.php

```diff
@@ -81,7 +81,10 @@ class_exists('Sonata\GoogleAuthenticator\GoogleAuthenticator')) {
         $loader->load('twig.xml');
         $loader->load('command.xml');
         $loader->load('actions.xml');
-        $loader->load('mailer.xml');
+
+        if ('sonata.user.mailer.default' === $config['mailer']) {
```

Comment:
Even if it shouldn't be a valid use case, I think we should check against the configured class too, since the service definition could be overridden by the user if I'm not wrong.

Reply:
As an alternative we could make the service non-public. It would get optimised out of the container if it was unused.

Reply:
You're right, I guess. But that would a BC break AFAIK.

Reply:
> I think we should check against the configured class too

It is possible that someone has injected the default mailer into something else, as well as replacing the service definition. I'm thinking this is going to be a BC break however we look at it. Should I rebase on master and make the service non-public rather than not loading the service definition? Feels cleaner to me.